src/transport/datamodel/utransport.rs

```diff
@@ -11,19 +11,66 @@
 
-use crate::transport::datamodel::{UAttributes, UListener, UPayload, UStatus};
-use crate::uri::datamodel::{UEntity, UUri};
+use async_trait::async_trait;
+use std::sync::mpsc::Sender;
 
+use crate::uprotocol::{UAttributes, UEntity, UMessage, UPayload, UStatus, UUri, Uuid};
+
+/// `UTransport` is the uP-L1 interface that provides a common API for uE developers to send and receive messages.
+///
+/// Implementations of `UTransport` contain the details for connecting to the underlying transport technology and
+/// sending `UMessage` using the configured technology. For more information, please refer to
+/// [uProtocol Specification](https://github.com/eclipse-uprotocol/uprotocol-spec/blob/main/up-l1/README.adoc).
+#[async_trait]
 pub trait UTransport {
-    /// API to register the calling uE with the underlying transport implementation.
-    fn register(&self, uentity: UEntity, token: &[u8]) -> UStatus;
+    /// Authenticates with the underlying transport layer that the `uEntity` passed
+    /// matches the transport specific identity. This method requires a resolved `UUri`.
+    ///
+    /// # Arguments
+    /// * `uEntity` - Resolved `UEntity`.
+    ///
+    /// # Returns
+    /// Returns () on success, otherwise an Err(UStatus) with the appropriate failure information.
+    async fn authenticate(&self, entity: UEntity) -> Result<(), UStatus>;
 
-    /// Transmit UPayload to the topic using the attributes defined in UTransportAttributes.
-    fn send(&self, topic: UUri, payload: UPayload, attributes: UAttributes) -> UStatus;
+    /// Transmits `UPayload` to the topic using the attributes defined in `UTransportAttributes`.
+    ///
+    /// # Arguments
+    /// * `topic` - Resolved `UUri` topic to send the payload to.
+    /// * `payload` - Actual payload.
+    /// * `attributes` - Additional transport attributes.
+    ///
+    /// # Returns
+    /// Returns () on success, otherwise an Err(UStatus) with the appropriate failure information.
+    async fn send(
+        &self,
+        topic: UUri,
+        payload: UPayload,
+        attributes: UAttributes,
+    ) -> Result<(), UStatus>;
 
-    /// Register a method that will be called when a message comes in on the specific topic.
-    fn register_listener(&self, topic: UUri, listener: dyn UListener) -> UStatus;
+    /// Registers a listener to be called when `UPayload` is received for the specific topic.
+    ///
+    /// # Arguments
+    /// * `topic` - Resolved `UUri` for where the message arrived via the underlying transport technology.
+    /// * `listener` - The method to execute to process the data for the topic.
+    ///
+    /// # Returns
+    /// Returns a Uuid on success that can be used for unregistering later, otherwise an Err(UStatus) with the appropriate failure information.
+    async fn register_listener(
+        &self,
+        topic: UUri,
+        listener: Sender<UMessage>,
+    ) -> Result<Uuid, UStatus>;
```

Comment:
FMPOV the returned Ok value should not be required to be a Uuid. The underlying implementation should be able to determine if a UUID is indeed required to keep it unique within its scope. For the MQTT based ulink, a simple MQTT 5 Subscription Identifier will probably be sufficient.
I would therefore suggest to change the type from `Uuid` to `String`. An implementation can then still return a (serialized) UUID if it so wishes.

@Mallets I have experimented with the following while playing around with an MQTT5 based implementation:
```rust
async fn register_listener<T>(&self, topic: UUri, listener: T) -> Result<String, UStatus>
    where
        T: Fn(UMessage) + Send + 'static;
```
WDYT?

Reply:
done

Reply:
done